### PR TITLE
Make checker stub files input to gradle java compile

### LINKED ISSUE #1300: Checker stub files (`config/checker/`) should be considered an input to Gradle JavaCompile tasks
So that when stubs change, the inputs to compilation are considered changed.

### PR DESCRIPTION
**Pull request by LukaszKontowski**
Closes #1300

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -139,6 +141,9 @@ allprojects {
     // `options.release = X` makes sure that regardless of Java version used to run the compiler,
     // only Java X-compatible APIs are available to the compiled code.
     options.release.set(Integer.parseInt(targetJavaVersion.majorVersion))
+
+    // Add files from config/checker directory as inputs to java compilation (so that changes trigger recompilation)
```
Comment (mkondratek) [inline build.gradle.kts:145]: hmmm, but what about kotlin compile? is it enough to add it here only?

Comment (LukaszKontowski) [reply build.gradle.kts:145]: It seems to me that the Checker Framework is for Java only? Also connected issue #1300 suggests that only JavaCompile should be affected. To be sure about this, @PawelLipski please refer.

Comment (PawelLipski) [reply build.gradle.kts:145]: :yes:, Checker is for Java exclusively (and hence I'm reluctant to increase the presence of Kotlin in this project, given our reliance on Checker)

File: build.gradle.kts [hunk 2]
```diff
@@ -63,6 +63,8 @@ tasks.register("printSignedPluginZipPath") {
   }
 }
 
+val configCheckerDirectory = rootProject.file("config/checker")
```
Comment (PawelLipski) [inline build.gradle.kts:66]: Nit: pls wrap into an `extra(...)` as stuff above is, and refer from CheckerFramework.kt in `-Astubs=` flag

File: build.gradle.kts (hunk 1 above)
Comment (PawelLipski) [reply build.gradle.kts:145]: Pls mention that in the code comment actually for clarity ☝🏻

**Review by mkondratek — APPROVED** [review]
ok, my questions have been answered. lgtm but pls apply Pawel's comments

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**